solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQParserPlugin.java

```diff
@@ -50,7 +51,21 @@
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
   @Override
-  public void init(@SuppressWarnings("rawtypes") NamedList args) {}
+  public void init(@SuppressWarnings("rawtypes") NamedList args) {
+    LTRThreadModule.maxThreads  = getInt(args.get("LTRMaxThreads"), LTRThreadModule.DEFAULT_MAX_THREADS, "LTRMaxThreads");
+    LTRThreadModule.maxQueryThreads = getInt(args.get("LTRMaxQueryThreads"), LTRThreadModule.DEFAULT_MAX_QUERYTHREADS, "LTRMaxQueryThreads");
+  }
+  private int getInt(Object thObj, int defValue, String paramName) throws NumberFormatException{
+     if (thObj != null) {
+       try{
+          return Integer.parseInt(thObj.toString());
+       }catch(NumberFormatException nfe){
+          String errorStr = nfe.toString() + ":" + paramName + " not an integer";
+          throw new NumberFormatException(errorStr);
```

Comment:
Do you have a unittest that verifies this happens if someone passes in a garbage thread count?